File: src/main/java/toker/panel/repository/CraftingRecipeItemRequirementRepository.java
package toker.panel.repository;

import org.springframework.data.jpa.repository.Modifying;
import toker.panel.entity.CraftingRecipeItemRequirement;

import javax.transaction.Transactional;

public interface CraftingRecipeItemRequirementRepository
        extends BaseRepository<CraftingRecipeItemRequirement, Integer> {

    @Transactional
    @Modifying
    void deleteAllByCraftingRecipeId(Integer recipeId);
}

File: src/main/java/toker/panel/repository/CraftingRecipeItemRequirementRepository.kt
package toker.panel.repository

import org.springframework.data.jpa.repository.Modifying
import toker.panel.entity.CraftingRecipeItemRequirement
import javax.transaction.Transactional

interface CraftingRecipeItemRequirementRepository : BaseRepository<CraftingRecipeItemRequirement, Int> {
    @Transactional
    @Modifying
    fun deleteAllByCraftingRecipeId(recipeId: Int)
}